src/vmm/src/builder.rs

```diff
@@ -811,16 +811,16 @@ pub fn configure_system_for_boot(
         cpu_config,
     };
 
-    // Configure vCPUs with normalizing and setting the generated CPU configuration.
-    for vcpu in vcpus.iter_mut() {
-        vcpu.kvm_vcpu
-            .configure(vmm.guest_memory(), entry_addr, &vcpu_config)
-            .map_err(VmmError::VcpuConfigure)
-            .map_err(Internal)?;
-    }
-
     #[cfg(target_arch = "x86_64")]
     {
+        // Configure vCPUs with normalizing and setting the generated CPU configuration.
+        for vcpu in vcpus.iter_mut() {
+            vcpu.kvm_vcpu
+                .configure(vmm.guest_memory(), entry_addr, &vcpu_config)
```

Comment:
nit: can also have `optional_capabilities` on x86_64, but simply have the struct be empty, so that the signatures of this function doesn't diverge?

Reply:
I don't think this is an issue here since the configuration blocks are already quite different. I would prefer to go opposite direction and actually move these arch specific blocks into some arch specific locations. But this is for a later time.